backend/app/usecase/search/search.go

```diff
@@ -22,12 +21,32 @@ type Search struct {
 
 // Result represents the result of a search query.
 type Result struct {
+	Resources ResourceResult
+	Err       error
+}
+
+// ResourceResult represents the resources obtained from a search query.
+type ResourceResult struct {
 	ShortLinks []entity.ShortLink
 	Users      []entity.User
 }
 
+// ErrUnknownResource represents unknown search resource error.
+type ErrUnknownResource struct{}
+
+func (e ErrUnknownResource) Error() string {
+	return "unknown resource"
+}
+
+// ErrUserNotProvided represents user not provided for search query.
+type ErrUserNotProvided struct{}
+
+func (e ErrUserNotProvided) Error() string {
+	return "user not provided"
+}
+
 // Search finds resources based on specified criteria.
-func (s Search) Search(query Query, filter Filter) (Result, error) {
+func (s Search) Search(query Query, filter Filter) (ResourceResult, error) {
```

Comment:
We are already returning error here. I wonder why are we putting extract error in `ResourceResult`. Go is making error an interface. By design, the language wants us to return whatever error we get directly.